File: math/GCD/go/gcd.go
package gcd

func gcd(a, b int) int {
	if a == 0 || b == 0 {
		return 0
	}

	if a == b {
		return a
	}

	if a > b {
		return gcd(a-b, b)
	}

	return gcd(a, b-a)
}

File: math/GCD/JavaScript/gcd.js
//recursive version of the euclidean algorithm

function gcd(a, b) {
	if (b==0) {
		return a;
	}
	else {
		return gcd(b, a%b);
	}
}

//tests

console.log(gcd(12,8));
//4

console.log(gcd(54,24));
//6

console.log(gcd(48,180));
//12

console.log(gcd(3,5));
//1
